src/platforms/hosted/windows/ftdi.c

```diff
@@ -0,0 +1,188 @@
+/*
+ * This file is part of the Black Magic Debug project.
+ *
+ * Copyright (C) 2011  Black Sphere Technologies Ltd.
+ * Copyright (C) 2018  Uwe Bonnes(bon@elektron.ikp.physik.tu-darmstadt.de)
+ * Written by Gareth McMullin <gareth@blacksphere.co.nz>
+ * Copyright (C) 2022-2023 1BitSquared <info@1bitsquared.com>
+ * Originated by Sid Price <sid@sidprice.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "general.h"
+#include "ftdi.h"
+#include "ftd2xx.h"
+#include "bmp_hosted.h"
+
+FT_HANDLE ftHandle;
+
+/*
+	The following structure is mocked by this module. FTD2XX
+	does not provide a similar structure.
+	
+	Values mocked:
+		ftdi_ctx.type	- The type of FTDI chip in the adapter
+*/
+static ftdi_context_s ftdi_ctx = {0}; // Used to fake libusb context and pass required parameters back to caller
+
+/*
+	The following array is used to map FTD2XX device type identifiers
+	to libftdi identifiers. The array is ordered by the FTD2XX
+	values, with the array entries being the libftdi values
+*/
+
+int ftdi_chip_types[] = {TYPE_AM, TYPE_BM,
+	-1, // FT_DEVICE_100AX not supported
+	-2, // Unknown type
+	TYPE_2232C, TYPE_R, TYPE_2232H, TYPE_4232H, TYPE_232H, TYPE_230X};
```

Comment:
Please add a ',' after TYPE_230X - that will make clang-format also fix the layout it's performed on this